src/app/controllers/auth.php

```diff
@@ -572,8 +575,18 @@ function get_magic_user() {
     }
 
     function login() {
+        if (!empty($_GET['next'])) {
+            $next_url = $_GET['next'];
```

Comment:
This will let someone use a URL that starts with `https://felicity.iiit.ac.in/` to redirect to any page they want after login right? Could be misleading. How about redirecting to `base_url()/` followed by the next part?

Also, what about URL encoding?